code/game/jobs/job/marines.dm

```diff
@@ -11,16 +11,25 @@
 	skills_type = /datum/skills/pfc
 	idtype = /obj/item/card/id/dogtag
 
-/datum/job/marine
-
-	generate_entry_message(mob/living/carbon/human/H)
-		if(H.assigned_squad)
-			. = "You have been assigned to: <b><font size=3 color=[squad_colors[H.assigned_squad.color]]>[lowertext(H.assigned_squad.name)] squad</font></b>.[flags_startup_parameters & ROLE_ADD_TO_MODE ? " Make your way to the cafeteria for some post-cryosleep chow, and then get equipped in your squad's prep room." : ""]"
+/datum/job/marine/generate_entry_message(mob/living/carbon/human/H)
+	if(H.assigned_squad)
+		. = "You have been assigned to: <b><font size=3 color=[squad_colors[H.assigned_squad.color]]>[lowertext(H.assigned_squad.name)] squad</font></b>.[flags_startup_parameters & ROLE_ADD_TO_MODE ? " Make your way to the cafeteria for some post-cryosleep chow, and then get equipped in your squad's prep room." : ""]"
 
-	generate_entry_conditions(mob/living/carbon/human/H)
-		. = ..()
-		if(flags_startup_parameters & ROLE_ADD_TO_MODE) H.nutrition = rand(60,250) //Start hungry for the default marine.
+/datum/job/marine/generate_entry_conditions(mob/living/carbon/human/H)
+	. = ..()
+	if(flags_startup_parameters & ROLE_ADD_TO_MODE) H.nutrition = rand(60,250) //Start hungry for the default marine.
 
+/datum/job/marine/generate_wearable_equipment()
+	if(MALE)
+		. = list(
```

Comment:
You'll might want to add it to a list `+=` instead of overriding `=`.

Reply:
I think i need to put += on every other instance actaully, since this is the parent.

Reply:
Procs get overridden when redefined for more specific objects.
`/proc/do()` won't be called if you have `/proc/mob/do()`, if `do()` is being called by the mob object, and it contains no `..()`

There's several ways you could do this.

Reply:
`/datum/job/marine/leader/generate_wearable_equipment()`, for example, could use a `. = ..()` at the beginning to add the towels to the list it will output.
Does the logic of this make sense to you?
You have two lists that will deal with each job, one for everyone and one for each specific case. Only one of them will be called, but that one can call the other. You want both lists to be output, so the one which will be called (the job-specific one) has to call the general one, get that list, and then add the specific contents to it before outputting them.